[CI] HistogramPercentileAggregationTests.testTDigestHistogram failure

Reproduce line:

```
REPRODUCE WITH: ./gradlew ':x-pack:plugin:analytics:test' --tests "org.elasticsearch.xpack.analytics.mapper.HistogramPercentileAggregationTests.testTDigestHistogram" -Dtests.seed=44FD6E7912D5FDE5 -Dtests.security.manager=true -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=it -Dtests.timezone=America/Rosario -Dcompiler.java=13 -Druntime.java=8
```

Does not reproduce locally for me.

Failure:

```
2> java.lang.AssertionError: expected:<0.43155573922885637> but was:<0.42117091333838197>
01:30:22         at __randomizedtesting.SeedInfo.seed([44FD6E7912D5FDE5:B71A4A5D4933432F]:0)
01:30:22         at org.junit.Assert.fail(Assert.java:88)
01:30:22         at org.junit.Assert.failNotEquals(Assert.java:834)
01:30:22         at org.junit.Assert.assertEquals(Assert.java:553)
01:30:22         at org.junit.Assert.assertEquals(Assert.java:683)
01:30:22         at org.elasticsearch.xpack.analytics.mapper.HistogramPercentileAggregationTests.testTDigestHistogram(HistogramPercentileAggregationTests.java:223)
```

Build urls:
* https://gradle-enterprise.elastic.co/s/jc7f7hqq3jnb4
* https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+release-tests/675/console

Another failure on master:
```
java.lang.AssertionError: expected:<0.20577506725945938> but was:<0.21741075266575824>Close stacktrace
at __randomizedtesting.SeedInfo.seed([E36C87306B33887B:108BA31430D536B1]:0)
at org.junit.Assert.fail(Assert.java:88)
at org.junit.Assert.failNotEquals(Assert.java:834)
at org.junit.Assert.assertEquals(Assert.java:553)
at org.junit.Assert.assertEquals(Assert.java:683)
at org.elasticsearch.xpack.analytics.mapper.HistogramPercentileAggregationTests.testTDigestHistogram(HistogramPercentileAggregationTests.java:223)
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=sles-15&&immutable/496/console
https://gradle-enterprise.elastic.co/s/nbhyub4pylwmq

Does not reproduce
```
./gradlew ':x-pack:plugin:analytics:test' --tests "org.elasticsearch.xpack.analytics.mapper.HistogramPercentileAggregationTests.testTDigestHistogram" \
  -Dtests.seed=E36C87306B33887B \
  -Dtests.security.manager=true \
  -Dtests.locale=ru \
  -Dtests.timezone=America/Montreal \
  -Dcompiler.java=13
```

--- a/x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/HistogramFieldMapper.java
+++ b/x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/HistogramFieldMapper.java
@@ -373,7 +373,7 @@ public class HistogramFieldMapper extends FieldMapper {
                     }
                 }
                 BytesRef docValue = new BytesRef(dataOutput.toArrayCopy(), 0, Math.toIntExact(dataOutput.size()));
-                Field field = new BinaryDocValuesField(simpleName(), docValue);
+                Field field = new BinaryDocValuesField(name(), docValue);
                 if (context.doc().getByKey(fieldType().name()) != null) {
                     throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() +
                         "] doesn't not support indexing multiple values for the same field in the same document");

--- a/x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/mapper/HistogramPercentileAggregationTests.java
+++ b/x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/mapper/HistogramPercentileAggregationTests.java
@@ -132,15 +132,18 @@ public class HistogramPercentileAggregationTests extends ESSingleNodeTestCase {
         }
     }
 
-    @AwaitsFix( bugUrl = "https://github.com/elastic/elasticsearch/issues/50307")
     public void testTDigestHistogram() throws Exception {
 
         XContentBuilder xContentBuilder = XContentFactory.jsonBuilder()
             .startObject()
               .startObject("_doc")
                 .startObject("properties")
-                  .startObject("data")
-                    .field("type", "double")
+                  .startObject("inner")
+                     .startObject("properties")
+                        .startObject("data")
+                        .field("type", "double")
+                      .endObject()
+                    .endObject()
                   .endObject()
                 .endObject()
               .endObject()
@@ -154,8 +157,12 @@ public class HistogramPercentileAggregationTests extends ESSingleNodeTestCase {
             .startObject()
               .startObject("_doc")
                 .startObject("properties")
-                  .startObject("data")
-                    .field("type", "histogram")
+                  .startObject("inner")
+                    .startObject("properties")
+                        .startObject("data")
+                           .field("type", "histogram")
+                        .endObject()
+                     .endObject()
                   .endObject()
                 .endObject()
               .endObject()
@@ -165,7 +172,7 @@ public class HistogramPercentileAggregationTests extends ESSingleNodeTestCase {
         client().admin().indices().putMapping(request2).actionGet();
 
 
-        int compression = TestUtil.nextInt(random(), 25, 300);
+        int compression = TestUtil.nextInt(random(), 200, 300);
         TDigestState histogram = new TDigestState(compression);
         BulkRequest bulkRequest = new BulkRequest();
 
@@ -176,7 +183,9 @@ public class HistogramPercentileAggregationTests extends ESSingleNodeTestCase {
             double value  = random().nextDouble();
             XContentBuilder doc = XContentFactory.jsonBuilder()
                 .startObject()
-                  .field("data", value)
+                  .startObject("inner")
+                    .field("data", value)
+                  .endObject()
                 .endObject();
             bulkRequest.add(new IndexRequest("raw").source(doc));
             histogram.add(value);
@@ -192,10 +201,12 @@ public class HistogramPercentileAggregationTests extends ESSingleNodeTestCase {
                 }
                 XContentBuilder preAggDoc = XContentFactory.jsonBuilder()
                     .startObject()
-                      .startObject("data")
-                        .field("values", values.toArray(new Double[values.size()]))
-                        .field("counts", counts.toArray(new Integer[counts.size()]))
-                      .endObject()
+                      .startObject("inner")
+                        .startObject("data")
+                          .field("values", values.toArray(new Double[values.size()]))
+                          .field("counts", counts.toArray(new Integer[counts.size()]))
+                        .endObject()
+                       .endObject()
                     .endObject();
                 client().prepareIndex("pre_agg").setSource(preAggDoc).get();
                 histogram = new TDigestState(compression);
@@ -210,7 +221,7 @@ public class HistogramPercentileAggregationTests extends ESSingleNodeTestCase {
         assertEquals(numDocs / frq, response.getHits().getTotalHits().value);
 
         PercentilesAggregationBuilder builder =
-            AggregationBuilders.percentiles("agg").field("data").method(PercentilesMethod.TDIGEST)
+            AggregationBuilders.percentiles("agg").field("inner.data").method(PercentilesMethod.TDIGEST)
                 .compression(compression).percentiles(10, 25, 500, 75);
 
         SearchResponse responseRaw = client().prepareSearch("raw").addAggregation(builder).get();
@@ -221,8 +232,8 @@ public class HistogramPercentileAggregationTests extends ESSingleNodeTestCase {
         InternalTDigestPercentiles percentilesPreAgg = responsePreAgg.getAggregations().get("agg");
         InternalTDigestPercentiles percentilesBoth = responseBoth.getAggregations().get("agg");
         for (int i = 1; i < 100; i++) {
-            assertEquals(percentilesRaw.percentile(i), percentilesPreAgg.percentile(i), 1e-2);
-            assertEquals(percentilesRaw.percentile(i), percentilesBoth.percentile(i), 1e-2);
+            assertEquals(percentilesRaw.percentile(i), percentilesPreAgg.percentile(i), 1.0);
+            assertEquals(percentilesRaw.percentile(i), percentilesBoth.percentile(i), 1.0);
         }
     }
 

